pyvisgen/simulation/data_set.py

```diff
@@ -197,5 +237,51 @@ def test_opts(rc):
     return min(active_telescopes_0, active_telescopes_1)
 
 
+def calc_ref_elev(src_crd, time, array_layout):
+    if time.shape == ():
+        time = time[None]
+    # Calculate for all times
+    # calculate GHA, Greenwich as reference for EHT
+    ha_all = Angle(
+        [t.sidereal_time("apparent", "greenwich") - src_crd.ra for t in time]
+    )
+
+    # calculate elevations
+    el_st_all = src_crd.transform_to(
+        AltAz(
+            obstime=time.reshape(len(time), -1),
+            location=EarthLocation.from_geocentric(
```

Comment:
It would also be faster do define the EarthLocations outside of this method since they don't change with time

Reply:
The source elevation changes over time, so we need that to be inside this method